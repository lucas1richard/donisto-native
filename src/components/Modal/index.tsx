import React from 'react';
import { Modal, View, ModalBaseProps, ScrollView } from 'react-native';
import Stack from 'components/StackHeader';
import logger from 'utilities/logger';
import StatusBarComponent from 'components/StatusBar';

interface RNModalProps extends ModalBaseProps {
  onShow?: () => any,
  onHide: () => any,
  children: any,
  title: string,
  onRequestClose?: () => any
}

class RNModal extends React.PureComponent<RNModalProps> {
  static marginTop: number;

  componentWillMount() {
    if (!RNModal.marginTop) {
      RNModal.marginTop = 20;
    }
  }

  modalWillShow = () => {
    if (this.props.onShow) {
      this.props.onShow();
    }
  }

  modalWillHide = () => {
    if (this.props.onRequestClose) {
      this.props.onRequestClose();
    }
    if (this.props.onHide) {
      this.props.onHide();
    }
  }

  render() {
    logger.log(RNModal.marginTop, 'Modal');
    const { children, title, ...rest } = this.props;
    return (
      <Modal animationType="slide" {...rest} onShow={this.modalWillShow}>
        <View style={{ flex: 1 }}>
          <StatusBarComponent />
          <Stack
            title={title}
            noBackOption
            onClose={this.modalWillHide}
          />
          <ScrollView>
            {children}
          </ScrollView>
        </View>
      </Modal>
    );
  }
}

export default RNModal;
