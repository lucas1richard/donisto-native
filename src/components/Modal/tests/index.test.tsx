import * as React from 'react';
import { shallow } from 'enzyme';
import Modal, { IRNModalProps } from '..';

describe('<Modal />', () => {
  const renderComponent = (props: IRNModalProps) => shallow(<Modal {...props} />);
  it('renders', () => {
    expect(renderComponent({
      onHide: jest.fn(),
      children: 'mockChildren'
    })).toMatchSnapshot();
  });
});