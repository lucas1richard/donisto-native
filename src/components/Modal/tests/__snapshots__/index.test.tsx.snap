// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RNModal
    onHide={[MockFunction]}
  >
    mockChildren
  </RNModal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animationType": "slide",
      "children": <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <StatusBarComponent />
        <StackHeader
          noBackOption={true}
          onClose={[Function]}
          title=""
        />
        <ScrollViewMock>
          mockChildren
        </ScrollViewMock>
      </Component>,
      "hardwareAccelerated": false,
      "onHide": [MockFunction],
      "onShow": [Function],
      "visible": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBarComponent />,
          <StackHeader
            noBackOption={true}
            onClose={[Function]}
            title=""
          />,
          <ScrollViewMock>
            mockChildren
          </ScrollViewMock>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "noBackOption": true,
            "onClose": [Function],
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "mockChildren",
          },
          "ref": null,
          "rendered": "mockChildren",
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animationType": "slide",
        "children": <Component
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <StatusBarComponent />
          <StackHeader
            noBackOption={true}
            onClose={[Function]}
            title=""
          />
          <ScrollViewMock>
            mockChildren
          </ScrollViewMock>
        </Component>,
        "hardwareAccelerated": false,
        "onHide": [MockFunction],
        "onShow": [Function],
        "visible": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StatusBarComponent />,
            <StackHeader
              noBackOption={true}
              onClose={[Function]}
              title=""
            />,
            <ScrollViewMock>
              mockChildren
            </ScrollViewMock>,
          ],
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "noBackOption": true,
              "onClose": [Function],
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "mockChildren",
            },
            "ref": null,
            "rendered": "mockChildren",
            "type": [MockFunction],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
